.github/templates/linux_binary_build_workflow.yml.j2

```diff
@@ -52,19 +52,31 @@ env:
 !{{ common.concurrency(build_environment) }}
 
 jobs:
+  get-label-type:
+    name: get-label-type
+    uses: ./.github/workflows/_runner-determinator.yml
+    with:
+      triggering_actor: ${{ github.triggering_actor }}
+      issue_owner: ${{ github.event.pull_request.user.login || github.event.issue.user.login }}
+      curr_branch: ${{ github.head_ref || github.ref_name }}
+      curr_ref_type: ${{ github.ref_type }}
+
 {%- for config in build_configs %}
   !{{ config["build_name"] }}-build:
+    needs: get-label-type
     if: ${{ github.repository_owner == 'pytorch' }}
     uses: ./.github/workflows/_binary-build-linux.yml
     with:!{{ upload.binary_env_as_input(config) }}
       {%- if "aarch64" in build_environment %}
-      runs_on: linux.arm64.m7g.4xlarge
+      runs_on: "${{ needs.get-label-type.outputs.label-type }}linux.arm64.m7g.4xlarge"
```

Comment:
I added a special runner_prefix input in https://github.com/pytorch/pytorch/pull/131826.

Let's use that instead (turns out the previous merge command hadn't worked unfortunately) since we won't generate the amzn2023 prefixes correctly without it